cmd/scrape.go

```diff
@@ -0,0 +1,543 @@
+    counter = counter + 1
+}
+
+func processBlocks(startBlock int, numBlocks int, skip int, nBlockProcesses int, nAddressProcesses int) {
+
+	blockChannel := make(chan int)
+	addressChannel := make(chan BlockInternals)
+
+	var blockWG sync.WaitGroup
+	blockWG.Add(nBlockProcesses)
+	for i := 0; i < nBlockProcesses; i++ {
+		go getTracesAndLogs(blockChannel, addressChannel, &blockWG)
+	}
+
+	var addressWG sync.WaitGroup
+	addressWG.Add(nAddressProcesses)
+	for i := 0; i < nAddressProcesses; i++ {
+		go extractAddresses(addressChannel, &addressWG)
+	}
+
+	for block := startBlock; block < startBlock+numBlocks; block = block + skip {
+		blockChannel <- block
+	}
+
+	close(blockChannel)
+	blockWG.Wait()
+
+	close(addressChannel)
+	addressWG.Wait()
+}
+
+func padLeft(str string, totalLen int) string {
+	if len(str) >= totalLen {
+		return str
+	}
+	zeros := ""
+	for i := 0; i < totalLen-len(str); i++ {
+		zeros += "0"
+	}
+	return zeros + str
+}
+
+// goodAddr Returns true if the address is not a precompile and not zero
+func goodAddr(addr string) bool {
+	// As per EIP 1352, all addresses less than the following value are reserved
+	// for pre-compiles. We don't index precompiles.
+	if addr < "0x000000000000000000000000000000000000ffff" {
+//	if addr < "0x0000000000000000000000000000000000000009" {
+		return false
+	}
+	return true
+}
+
+// potentialAddress Processing 'input' value, 'output' value or event 'data' value
+// we do our best, but we don't include everything we could. We do the best we can
+func potentialAddress(addr string) bool {
+	// Any address smaller than this we call a 'baddress' and do not index
+	small := "00000000000000000000000000000000000000ffffffffffffffffffffffffff"
+	//        -------+-------+-------+-------+-------+-------+-------+-------+
+	if addr <= small {
+		return false
+	}
+
+	// Any address with less than this many leading zeros is not an left-padded 20-byte address
+	largePrefix := "000000000000000000000000"
+	//              -------+-------+-------+
+	if !strings.HasPrefix(addr, largePrefix) {
+		return false
+	}
+
+	if strings.HasSuffix(addr, "00000000") {
+		return false
+	}
+	return true
+}
+
+var scrapeCmd = &cobra.Command{
+	Use:   "scrape",
+	Short: "Freshen the index to the front of the chain",
+	Long: `
+Description:
+
+  The 'scrape' subcommand freshens the TrueBlocks index, picking up where it last
+  left off. 'Scrape' visits every block, queries that block's traces and logs
+  looking for addresses, and writes an index of those addresses per transaction.
+  This allows for lightning fast querying of transaction histories, something
+  that is not practically possible directly against the node.`,
+	Run: func(cmd *cobra.Command, args []string) {
+		start := viper.GetInt("startBlock")
+		if start < 0 {
+			start = 1
+		}
+		n := viper.GetInt("nBlocks")
+		skip := 1
+		blockChans := viper.GetInt("nBlockChans")
```

Comment:
These are named wrong; should be nBlockProcesses, not nBlockChans.

Reply:
Changed.  Question for you. Do you think command line options should be `--dash-separated` or `--camelCase`?

Reply:
Not sure! We should look at some other Go projects and copy.